### PR TITLE
remove the txid from the committer list after the listeners have been called

### PR DESCRIPTION
**Pull request by arner**
Otherwise the list keeps growing, which causes SQL failures after 1000 transactions and possibly other issues as well.

### CODE REVIEW

File: platform/common/core/generic/committer/finality.go [hunk 1]
```diff
@@ -110,21 +116,25 @@ func (c *FinalityManager[V]) runStatusListener(ctx context.Context) {
...
 			txIDs := c.txIDs.ToSlice()
 			c.mutex.RUnlock()
 			if len(txIDs) == 0 {
-				c.logger.Debugf("no transactions to check vault status")
+				// c.logger.Debugf("no transactions to check vault status")
 				break
 			}
 
 			newCtx, span := c.tracer.Start(context.Background(), "vault_status_check")
 			c.logger.Debugf("check vault status for [%d] transactions", len(txIDs))
```
Comment (adecaro) [inline platform/common/core/generic/committer/finality.go:131]: do you want to remove this?

File: platform/common/core/generic/committer/finality.go [hunk 2]
```diff
@@ -97,9 +99,13 @@ func (c *FinalityManager[V]) runEventQueue(context context.Context) {
 	for {
 		select {
 		case <-context.Done():
+			c.logger.Debugf("closing event queue")
 			return
 		case event := <-c.eventQueue:
 			c.listenerManager.InvokeListeners(event)
+			c.mutex.Lock()
```
Comment (adecaro) [inline platform/common/core/generic/committer/finality.go:106]: this makes me think about handling panics. if InvokeListeners panics, then we don't dispatch events anymore.

**Review by adecaro — COMMENTED** [review]
@arne, please, can you check the tests?

**Comment by adecaro** [comment]
@arner , please, can you test this commit with token-sdk?

File: platform/common/core/generic/committer/finality.go [hunk 3]
```diff
@@ -80,6 +81,7 @@ func (c *FinalityManager[V]) RemoveListener(txID driver.TxID, toRemove driver.Fi
 	defer c.mutex.Unlock()
 	c.txIDs.Remove(txID)
 	c.listenerManager.RemoveListener(txID, toRemove)
+	// c.logger.Debugf("removed listener for [%s]: now listening to %d transactions...", txID, c.txIDs.Length())
```
Comment (alexandrosfilios) [inline platform/common/core/generic/committer/finality.go:84]: We can remove them or enable them?

**Comment by arner** [comment]
Superseded by https://github.com/hyperledger-labs/fabric-smart-client/pull/683

**Pull request closed without merging**